org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/translator/A4SolutionReader.java

```diff
@@ -354,9 +354,22 @@ private A4SolutionReader(Iterable<Sig> sigs, XMLNode xml) throws IOException, Er
         final int maxseq = Integer.parseInt(inst.getAttribute("maxseq"));
         final int max = Util.max(bitwidth), min = Util.min(bitwidth);
         if (bitwidth >= 1 && bitwidth <= 30)
-            for (int i = min; i <= max; i++) {
-                atoms.add(Integer.toString(i));
+            for(int i=min; i<=max; i++) { 
+                atoms.add(Integer.toString(i)); 
             }
+        // Inserting the exceeded integers into the atoms list
+        for (XMLNode i : inst){
+            if (i.getAttribute("label").equals("Int")) {
+                for (XMLNode y: i) {
+                    try{
+                        int value = Integer.valueOf(y.getAttribute("value"));
+                        atoms.add(String.valueOf(value));
+                    }catch (NumberFormatException e) {
+                        // throw new ErrorSyntax("The XML file must containt integer type instead of:" + y.getAttribute("value"));
```

Comment:
please handle this exception (same comment as for ignoring `NumberFormatException` about

Reply:
Good catch! fixed.